src/include/uapi/ipc/dai-intel.h

```diff
@@ -179,10 +187,13 @@ struct sof_ipc_dai_dmic_params {
 	uint16_t duty_min;	/**< Min. mic clock duty cycle in % (20..80) */
 	uint16_t duty_max;	/**< Max. mic clock duty cycle in % (min..80) */
 
+	uint32_t wake_up_time;      /**< Time from clock start to data (us) */
+	uint32_t min_clock_on_time; /**< Min. time that clk is kept on (us) */
+
 	uint32_t num_pdm_active; /**< Number of active pdm controllers */
```

Comment:
My very immature thought: what about to put definition in the end of the structure. If the variables is in the end, it will not mess the original memory positions of the old variable and old driver/fw can still use the old variables correctly when version unmatched. I know we have version check for the FW. But put the new variables in the end of the structure may be more stable even the version check doesn't work well. This is my rough thought and may be not correct. : - )

Reply:
Yep, it's a good idea. I'll change that.

Reply:
@libinyang Done!

Reply:
@singalsu can you add all new fields to the end of a structure, that way we preserve ABI.

Reply:
@lgirdwood I Do you mean that new parameters need to be after reserved[]?